src/Utils/signal.ts

```diff
@@ -81,27 +82,36 @@ export const parseAndInjectE2ESessions = async(
 		assertNodeErrorFree(node)
 	}
 
-	await Promise.all(
-		nodes.map(
-			async node => {
-				const signedKey = getBinaryNodeChild(node, 'skey')!
-				const key = getBinaryNodeChild(node, 'key')!
-				const identity = getBinaryNodeChildBuffer(node, 'identity')!
-				const jid = node.attrs.jid
-				const registrationId = getBinaryNodeChildUInt(node, 'registration', 4)
-
-				await repository.injectE2ESession({
-					jid,
-					session: {
-						registrationId: registrationId!,
-						identityKey: generateSignalPubKey(identity),
-						signedPreKey: extractKey(signedKey)!,
-						preKey: extractKey(key)!
-					}
-				})
-			}
+	// Most of the work in repository.injectE2ESession is CPU intensive, not IO
+	// So Promise.all doesn't really help here,
+	// but blocks even loop if we're using it inside keys.transaction, and it makes it "sync" actually
+	// This way we chunk it in smaller parts and between those parts we can yield to the event loop
+	// It's rare case when you need to E2E sessions for so many users, but it's possible
+	const chunkSize = 100
+	const chunks = chunk(nodes, chunkSize)
```

Comment:
I haven't tested it, but the idea is good!

Some unnecessary things, such as the lodash chunk, could use 
"chunk" of Array.from(), which can be done natively by ES6 javascript itself

Reply:
TY for the feedback!

> "chunk" of Array.from(), which can be done natively by ES6 javascript itself

Hah, didn't know about that, could you point to the reference for that? 
I didn't find "chunk" in the doc, not sure if it's the right reference tho
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from

> Some unnecessary things, such as the lodash chunk

The project already has lodash (dependencies use it) https://github.com/WhiskeySockets/Baileys/blob/master/yarn.lock#L5142-L5145
So I just added it explicitly in the dependencies (but would happy to use native ES6 feature for SURE!)

Reply:
TY for the feedback!

> "chunk" of Array.from(), which can be done natively by ES6 javascript itself

Hah, didn't know about that, could you point to the reference for that? 
I didn't find "chunk" in the doc, not sure if it's the right reference tho
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from

> Some unnecessary things, such as the lodash chunk

The project already has lodash (dependencies use it) https://github.com/WhiskeySockets/Baileys/blob/master/yarn.lock#L5142-L5145
So I just added it explicitly in the dependencies (but would happy to use native ES6 feature for SURE!)